File: Data Structures & Algorithms/find-duplicate-integer/submission-1.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n=nums.length;
        int c=nums[0];
        int temp;
        while(true){
            if(nums[c]>0){
                temp=nums[c];
                nums[c] = -1;
                c=temp;
            }else if(nums[c]==-1){
                return c;
            }
        }
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        return max(nums[0],self.dfs(nums[1:]),self.dfs(nums[:-1]))

    def dfs(self,nums):
        rob1,rob2 = 0,0
        for i in nums:
            newrob = max(i+rob1,rob2)
            rob1 = rob2
            rob2 = newrob 
        return rob2
        